instrument-starters/opentracing-spring-cloud-jdbc-starter/src/main/java/io/opentracing/contrib/spring/cloud/jdbc/JdbcAspect.java

```diff
@@ -52,10 +53,19 @@ public JdbcAspect(boolean withActiveSpanOnly, Set<String> ignoredStatements) {
   @Around("execution(java.sql.Connection *.getConnection(..)) && target(javax.sql.DataSource)")
   public Object getConnection(final ProceedingJoinPoint pjp) throws Throwable {
     Connection conn = (Connection) pjp.proceed();
-    if (conn instanceof TracingConnection ||
-        conn.isWrapperFor(TracingConnection.class)) {
-      return conn;
+    try {
+      if (conn instanceof TracingConnection ||
+          conn.isWrapperFor(TracingConnection.class)) {
+        return conn;
+      }
+    } catch (SQLException ignored) {
+      /**
+       * Oracle JDBC driver throws this SQL exception: "Object does not wrap anything with requested interface".
+       *
+       * See {@link java.sql.Wrapper#isWrapperFor(Class)} for an explanation why this should not be.
```

Comment:
The java doc for isWrapperFor states: _Returns true if this either implements the interface argument or is directly or indirectly a wrapper for an object that does._ By passing a class as an argument, this contract is broken. The solution here would be, to make TracingConnection an interface.
Best regards,
Lyca

Reply:
You're right, when I pass an interface the oracle connection does not throw an error.

So, close this one and open a new one for the TracingConnection?

Reply:
My idea about it would be to move/rename the current TracingConnection to TracingConnectionWrapper and let it implement the newly created TracingConnection which extends Connection. (Oracle is doing it like this in ojdbc)
So you are right, the [OpenTracing JDBC Instrumentation](https://github.com/opentracing-contrib/java-jdbc) needs to be fixed first, and then the PR here must incorporate those changes.